tools/similarity_search/test/index.spec.ts

```diff
@@ -1,22 +1,98 @@
 
 import "../src/index"
 
+interface SimilarityCheckResponse {
+  similarity_score: number
+}
+
 describe("Authentication", () => {
-  it("returns 401 Unauthorized when API key is missing or invalid", async () => {
+  it("returns 401 Unauthorized when API key is invalid", async () => {
     const response = await SELF.fetch("https://example.com/", {
       method: "POST",
       headers: {
-        "Content-Type": "application/json"
+        "Content-Type": "application/json",
+        "X-API-Key": "invalid-api-key",
       },
       body: JSON.stringify({
         text: "Sample text",
-        namespace: "test-namespace"
-      })
+        namespace: "test-namespace",
+      }),
     })
 
     expect(response.status).toBe(401)
     expect(await response.text()).toBe("Unauthorized")
   })
 })
+
+describe("Validation", () => {
+  it("returns 400 Invalid JSON format when text is missing", async () => {
+    const response = await SELF.fetch("https://example.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-Key": "test-api-key",
+      },
+      body: JSON.stringify({
+        namespace: "test-namespace",
+      }),
+    })
+
+    expect(response.status).toBe(400)
+    expect(await response.text()).toBe("Invalid JSON format")
+  })
+
+  it("returns 400 Invalid JSON format when namespace is missing", async () => {
+    const response = await SELF.fetch("https://example.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-Key": "test-api-key",
+      },
+      body: JSON.stringify({
+        text: "Sample text",
+      }),
+    })
+
+    expect(response.status).toBe(400)
+    expect(await response.text()).toBe("Invalid JSON format")
+  })
+})
+
+describe("Functionality", () => {
+  it("runs AI model and gets vectorized string back", async () => {
+    const response = await env.AI.run("@cf/baai/bge-base-en-v1.5", {
+      text: ["Sample text"],
+    })
+    expect(response).toHaveProperty("data")
+  })
+
+  it("queries vector database and gets proper response", async () => {
+    const response = await env.VECTORIZE_INDEX.query([1, 2, 3], {
+      namespace: "test-namespace",
+      topK: 1,
+    })
+    expect(response).toHaveProperty("matches")
+    expect(response.matches.length).toBeGreaterThan(0)
+  })
+
+  it("returns similarity score for valid requests", async () => {
+    const response = await SELF.fetch("https://example.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-API-Key": "test-api-key",
+      },
+      body: JSON.stringify({
+        text: "Sample text",
+        namespace: "test-namespace",
+      }),
+    })
+
+    expect(response.status).toBe(200)
+    const jsonResponse: SimilarityCheckResponse = await response.json()
+    expect(jsonResponse).toHaveProperty("similarity_score")
+    expect(typeof jsonResponse.similarity_score).toBe("number")
+  })
+})
```

Comment:
These look like unit tests to me. For example, you can't trigger Worker AI yourself - this function is an internal component of the worker you are testing. Same for querying vectorize. The only trigger at your disposal is user's request. You can, however, assert all other worker interactions, including worker AI being called and vector DB being querying.

Reply:
> These look like unit tests to me. 

That's what I thought initially when you told me that these services behavior should be tested too. But now I see what you are talking about. 

I changed it so now we are just looking at what AI and DB are called with and what they return during normal request to a service, instead of calling them directly. I don't know if this is sufficient, or the abstraction has to be even deeper.